qos-service-impl/src/main/java/com/palantir/atlasdb/qos/config/SimpleThrottlingStrategy.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.qos.config;
+
+import java.util.List;
+import java.util.function.Supplier;
+
+import com.palantir.atlasdb.qos.ratelimit.guava.RateLimiter;
+
+import net.jcip.annotations.GuardedBy;
+import net.jcip.annotations.ThreadSafe;
+
+@ThreadSafe
+public class SimpleThrottlingStrategy implements ThrottlingStrategy {
+    private static final double ONCE_EVERY_HUNDRED_SECONDS = 0.01;
+    private final RateLimiter rateLimiter;
+    @GuardedBy("this") private double multiplier;
+
+    public SimpleThrottlingStrategy() {
+        this.rateLimiter = RateLimiter.create(ONCE_EVERY_HUNDRED_SECONDS);
+        this.multiplier = 1.0;
+    }
+
+    @Override
+    public double getClientLimitMultiplier(Supplier<List<CassandraHealthMetricMeasurement>> metricMeasurements) {
+        if (shouldAdjust()) {
+            if (cassandraIsUnhealthy(metricMeasurements)) {
+                halveTheRateMultiplier();
+            } else {
+                // TODO(hsaraogi): increase only if the client is consuming its limit.
+                increaseTheRateMultiplier();
+            }
+        }
+        return multiplier;
```

Comment:
this is not necessarily a safe read per JMM as you're synchronizing the writes but not read of `multiplier`.

Reply:
I think we are fine if we have eventual consistency for the reads, did not synchronize the reads as reads are much more frequent than updates.